src/main/java/googlesignin/GoogleUserMapper.java

```diff
@@ -8,18 +12,23 @@
 import java.util.Base64;
 
 public final class GoogleUserMapper {
-    private static Base64.Encoder encoder;
 
-    private GoogleUserMapper() {
-        encoder = Base64.getEncoder();
-    }
+    @Autowired
+    private static UserAccessor userAccessor;
+
+    private GoogleUserMapper() { }
 
     public static String getUsernameFromEmail(String email) {
         int usernameEndsAt = email.indexOf('@');
         if(usernameEndsAt == -1) {
             throw new IllegalArgumentException();
         }
-        return email.substring(0, usernameEndsAt);
+        String baseUsername = email.substring(0, usernameEndsAt);
+        String username = baseUsername;
+        while(userAccessor.checkIfUsernameExists(username)) {
+            username = baseUsername + RandomStringUtils.randomAlphanumeric(3);
```

Comment:
Consider leaving a comment here about what this is doing.

Reply:
Added